pandas/tests/indexes/test_multi.py

```diff
@@ -2489,7 +2489,27 @@ def test_reconstruct_remove_unused(self):
         # idempotent
         result2 = result.remove_unused_levels()
         tm.assert_index_equal(result2, expected)
-        assert result2 is result
+        assert result2.is_(result)
+
+    def test_remove_unused_levels_large(self):
+        # because tests should be deterministic:
+        rng = np.random.RandomState(4)  # chosen by fair dice roll.
+
+        size = 1 << 16
+        df = DataFrame(dict(first=rng.randint(0, 1 << 13, size),
+                            second=rng.randint(0, 1 << 10, size),
+                            third=rng.rand(size)))
+        df = df.groupby(['first', 'second']).sum()
+        df = df[df.third < 0.1]
+
+        result = df.index.remove_unused_levels()
+        assert len(result.levels[0]) < len(df.index.levels[0])
+        assert len(result.levels[1]) < len(df.index.levels[1])
+        assert result.equals(df.index)
+
```

Comment:
also construct that this IS equal to ``.reset_index(..).set_index(...).index``. (for each example)